client/fingerprint/vault.go

```diff
@@ -0,0 +1,90 @@
+package fingerprint
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	client "github.com/hashicorp/nomad/client/config"
+	"github.com/hashicorp/nomad/nomad/structs"
+	vapi "github.com/hashicorp/vault/api"
+)
+
+const (
+	vaultAvailable   = "available"
+	vaultUnavailable = "unavailable"
+)
+
+// ConsulFingerprint is used to fingerprint the architecture
+type VaultFingerprint struct {
+	logger    *log.Logger
+	client    *vapi.Client
+	lastState string
+}
+
+// NewVaultFingerprint is used to create a Vault fingerprint
+func NewVaultFingerprint(logger *log.Logger) Fingerprint {
+	return &VaultFingerprint{logger: logger, lastState: vaultUnavailable}
+}
+
+func (f *VaultFingerprint) Fingerprint(config *client.Config, node *structs.Node) (bool, error) {
+	if config.VaultConfig == nil || !config.VaultConfig.Enabled {
+		return false, nil
+	}
+
+	// Only create the client once to avoid creating too many connections to
+	// Vault.
+	if f.client == nil {
+		vaultConfig, err := config.VaultConfig.ApiConfig()
+		if err != nil {
+			return false, fmt.Errorf("Failed to initialize the Vault client config: %v", err)
+		}
+
+		f.client, err = vapi.NewClient(vaultConfig)
+		if err != nil {
+			return false, fmt.Errorf("Failed to initialize Vault client: %s", err)
+		}
+	}
+
+	// Connect to vault and parse its information
+	status, err := f.client.Sys().SealStatus()
+	if err != nil {
+		// Clear any attributes set by a previous fingerprint.
+		f.clearVaultAttributes(node)
+
+		// Print a message indicating that Vault is not available anymore
+		if f.lastState == vaultAvailable {
+			f.logger.Printf("[INFO] fingerprint.consul: Vault is unavailable")
+		}
+		f.lastState = vaultUnavailable
+		return false, nil
+	}
+
+	node.Attributes["vault.accessible"] = strconv.FormatBool(true)
```

Comment:
If we are expecting a min version of the Vault API we should ensure that we do the check during fingerprinting.